src/main/java/com/aws/greengrass/util/platforms/unix/UnixPlatform.java

```diff
@@ -549,17 +549,27 @@ private List<String> children(String parent, Map<String, List<String>> procMap)
         return ret;
     }
 
-    private String getIpcServerSocketAbsolutePath(Path rootPath) {
-        return rootPath.resolve(IPC_SERVER_DOMAIN_SOCKET_FILENAME).toString();
+    private String getIpcServerSocketAbsolutePath(Path rootPath, String ipcPath) {
+        if (Utils.isEmpty(ipcPath)) {
+            ipcPath = IPC_SERVER_DOMAIN_SOCKET_FILENAME;
+        }
+        return rootPath.resolve(ipcPath).toString();
```

Comment:
If `ipcPath` is provided by the user, then that path is _not_ relative to the root path. The user-provided path should be used directly, without modification.